src/validation.cpp

```diff
@@ -2816,15 +2819,20 @@ bool CChainState::InvalidateBlock(CValidationState& state, const CChainParams& c
         setDirtyBlockIndex.insert(invalid_walk_tip);
         setBlockIndexCandidates.erase(invalid_walk_tip);
         setBlockIndexCandidates.insert(invalid_walk_tip->pprev);
+
+        // If we abort invalidation after this iteration, make sure
+        // the last disconnected block gets marked failed (rather than
+        // just child of failed)
+        to_mark_failed = invalid_walk_tip;
     }
 
     {
         // Mark the block itself as invalid.
```

Comment:
Commit ebe6eebab3b3a42da70ec99c1f30568ae0abb16f 

Update comment? I mean, can be a different block?

Reply:
Clarified the comment.